### PR TITLE
add & address character and avoid freeing up memory on the stack

### PR DESCRIPTION
**Pull request by Phoenix8215**

### CODE REVIEW

File: chapter5-tensorrt-api-basics/5.2-load-model/src/cpp/utils.hpp [hunk 1]
```diff
@@ -52,16 +52,20 @@ static void __log_info(Level level, const char* format, ...) {
...
 
     if (level == Level::INFO) {
-        n += snprintf(msg + n, sizeof(msg) - n, YELLOW "[info]:");
+        n += snprintf(msg + n, sizeof(msg) - n, YELLOW "[info]: " CLEAR);
     } else if (level == Level::VERB) {
-        n += snprintf(msg + n, sizeof(msg) - n, PURPLE "[verb]:" CLEAR);
+        n += snprintf(msg + n, sizeof(msg) - n, PURPLE "[verb]: " CLEAR);
     } else {
-        n += snprintf(msg + n, sizeof(msg) - n, RED    "[error]:");
+        n += snprintf(msg + n, sizeof(msg) - n, RED    "[verb]: " CLEAR);
```
Comment (kalfazed) [inline chapter5-tensorrt-api-basics/5.2-load-model/src/cpp/utils.hpp:59]: Why is it changed from `[error]` to `[verb]`?

File: chapter5-tensorrt-api-basics/5.3-infer-model/src/cpp/utils.hpp [hunk 2]
```diff
@@ -64,7 +64,6 @@ static void __log_info(Level level, const char* format, ...) {
     va_end(args);
 
     if (level == Level::ERROR) {
-        free(msg);
         exit(1);
```
Comment (kalfazed) [inline chapter5-tensorrt-api-basics/5.3-infer-model/src/cpp/utils.hpp:67]: Great! Thank you for helping me to find this unnecessary memory free for stack. Btw, I realized that may be it is not good to exit the program from the logger. We should create an `assert` to terminated the program with unexpected behavior. So my suggestion is that we also need to remove the `exit(1)` here.

File: chapter5-tensorrt-api-basics/5.4-print-structure/src/cpp/utils.hpp [hunk 3]
```diff
@@ -63,7 +63,6 @@ static void __log_info(Level level, const char* format, ...) {
     va_end(args);
 
     if (level == Level::ERROR) {
-        free(msg);
         exit(1);
```
Comment (kalfazed) [inline chapter5-tensorrt-api-basics/5.4-print-structure/src/cpp/utils.hpp:66]: same here.

**Review by kalfazed — COMMENTED** [review]
Hi @Phoenix8215, thank you for this PR. I left some comments and it will be helpful if you can also remove the unnecessary `exit(1)` from the logger.

**Comment by Phoenix8215** [comment]
It's very kindly for you to provide such repo.I think it's really usefully for the freshman to learn how to deploy dl models with TensorRT step by step🥰

**Comment by kalfazed** [comment]
The pleasure is mine. Thank you for the PR. I will keep on updating this repository in the future. Could you squash and merge it now?

**Comment by Phoenix8215** [comment]
> The pleasure is mine. Thank you for the PR. I will keep on updating this repository in the future. Could you squash and merge it now?

I don't have write permission .Maybe you can merge it🥲

**Pull request merged**